pages/apply/[resident]/your-situation.tsx

```diff
@@ -0,0 +1,262 @@
+        return PropertyOwnership;
+
+      case '9':
+        return SoldProperty;
+
+      case '10':
+        return RelationshipBreakdown;
+
+      case '11':
+        return Arrears;
+
+      case '12':
+        return UnderOccupying;
+
+      case '13':
+        return Benefits;
+
+      case '14':
+        return Landlord;
+
+      case '15':
+        return OtherHousingRegister;
+
+      case '16':
+        return BreachOfTenancy;
+
+      case '17':
+        return legalRestrictions;
+
+      case '18':
+        return unspentConvictions;
+
+      default:
+        return undefined;
+    }
+  };
+
+  const router = useRouter();
+  const store = useStore<Store>();
+
+  let thisResident = router.query.resident as string;
+  const currentResident = getResident(thisResident, store.getState());
+
+  let name = '';
+  if (currentResident && currentResident.name) {
+    name = currentResident.name;
+  }
+
+  const [applicationData, setApplicationData] = useState({});
+  const [initValues, setInitValues] = useState({});
+  const [questions, setQuestions] = useState({});
+  const [currentAccomodation, setCurrentAccomodation] = useState(
+    currentResident?.formData['address-details']?.['current-acommodation']
+  );
+  const [onExit, setOnExit] = useState(false);
+
+  const baseHref = `/apply/${currentResident?.slug}`;
+  const returnHref = '/apply/overview';
+  const breadcrumbs = [
+    {
+      href: returnHref,
+      name: 'Application',
+    },
+    {
+      href: baseHref,
+      name: `${currentResident?.name}`,
+    },
+    {
+      href: `${baseHref}/your-situation`,
+      name: `Your Situation`,
+    },
+  ];
+
+  const activeStep = 'your-situation';
+
+  if (Object.keys(questions).length === 0) {
+    console.log('Current Accomodation: ', currentAccomodation);
+
+    if (!currentAccomodation || currentAccomodation.length === 0) {
+      //router.push(returnHref);
+      setQuestions(questionData('1')!);
+    }
+
+    switch (currentAccomodation) {
+      case 'squatter':
+        setQuestions(questionData('1')!);
+        break;
+      case 'unauthorised-occupant':
+        setQuestions(questionData('2')!);
+        break;
+      case 'owner-occupier':
+        setQuestions(questionData('3')!);
+        break;
+      default:
+        setQuestions(questionData('1')!);
+        break;
+    }
+  }
+
+  const onSave = (values: FormData) => {
+    const data: { [key: string]: FormData } = { ...applicationData };
+
+    data[activeStep!] = values;
+
+    //console.log('the data', data);
+
+    const formData = data['your-situation'];
+    //console.log('formData', formData);
+
+    const stepName: string = questions.steps[0].fields[0].name;
+    //console.log('step Name', stepName);
+
+    const extractedValue: string = formData[stepName];
+    //console.log('extractedValue', extractedValue);
+
+    if (exitForm(extractedValue, stepName)) {
+      router.push(returnHref);
+    }
+
+    const nextQuestion = questionData(extractedValue);
```

Comment:
I don't think we should confuse the answer that the user gave with the next quesiton that they need to be directed to.

I'd prefer that the `nextQuestion` was a different value stored in the JSON data and that the answer to the original question was simply "yes" or "no" or "true" or "false" or whatever.

For example, instead of:
```
          "options": [
            {
              "label": "Yes",
              "value": "11"
            },
```

what about:
```
          "options": [
            {
              "label": "Yes",
              "value": "yes",
              "nextStep": 11
            },
```

or maybe even better:

```
          "options": [
            {
              "label": "Yes",
              "value": "yes",
              "nextStep": "court-order"
            },
```